### DISCUSSION TITLE
Help with installing Firedrake under WSL

### DISCUSSION THREAD

**Discussion by walkersw** [firedrakeproject/firedrake#2609] [Firedrake support] [answered]
I have installed Ubuntu though Windows WSL.  I did the install script, and most things seemed to work, but it crapped out at HDF5.  Is that really necessary?  Also, I tried running the firedrake-update, and it broke when installing petsc4py.  Any help?  I have included the log files.

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/9915393/firedrake-install.log)
[firedrake-update.log](https://github.com/firedrakeproject/firedrake/files/9915396/firedrake-update.log)
[configure.log](https://github.com/firedrakeproject/firedrake/files/9915399/configure.log)

**Comment by connorjward** [comment]
The only thing I can see in the logs that might be abnormal is the you are installing Firedrake into `/mnt/c` instead of somewhere in your *Linux* home directory. Our page on [installing for WSL](https://github.com/firedrakeproject/firedrake/wiki/Installing-on-Windows-Subsystem-for-Linux) mentions that `/mnt/c` can cause issues during the install.

Could you try again but install to `~/firedrake` or similar?

**Comment by walkersw** [comment, marked as answer]
Yes, I know about the issue with possible anaconda installation.  I searched the log file and found no mention of "conda".

I would rather not have Firedrake installed in that "root" directory, which is really under my C:\users\ directory.  Is there no way to make it install elsewhere?

**Reply by connorjward** [reply]
I don't know if it could work installed under `/mnt/c`. I just know that Windows has a different file system to Linux (e.g. case insensitive) and think that this might be causing your problem. It would at least be interesting to see if the installation works inside your home directory.

Why do you specifically want Firedrake to install there? Windows now has some really solid support for accessing WSL files easily through VS Code, File Explorer, Terminal etc.

**Reply by walkersw** [reply]
I'll try it the "correct" way.  I'll let you know if it works.

**Reply by walkersw** [reply]
Ok, I completely removed everything, even Ubuntu, and reinstalled Ubuntu under WSL.  I then unmounted my C drive (as per the instructions) and did the Firedrake install.  It crapped out at the same place (the log file says it fails at HDF5).

I tried to run it in python3 anyway.  After doing the "source" command, I go into python3 and type:

from firedrake import *

It gives a traceback error and says:

ModuleNotFoundError: No module named 'petsc4py'

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/9923124/firedrake-install.log)

**Reply by ksagiyam** [reply]
Your install did not complete; could you post `/home/walker/firedrake/src/petsc/configure.log`?

**Reply by walkersw** [reply]
Alright, I seem to have it working now.  I redid the installation with the --minimal-petsc flag.

There was also something where Windows had a window come up asking if I want to allow whatever is happening in the install script to access my home network and/or public networks.  This happened 3 separate times.  Before I accidentally limited it to home network.  When I installed with the minimal-petsc flag just now, I made sure to enable both home and public networks.  I don't know if this made a difference.

**Reply by walkersw** [reply]
Ok, I redid the installation, without needing minimal-petsc.  I made sure to click all networks to give access to the install.  Everything works now.
